### PR TITLE
chore: normalize md formatting

### PR DESCRIPTION
**Pull request by mujahidkay**
Closes: https://github.com/Agoric/documentation/issues/945

### Description

 - Adds a prettier config for formatting md files
 - Adds a `lint:format` and `format` script for yarn
 - Adds CI for `lint:format`
 - Formats all markdown files

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1 @@
+npx lint-staged
```
Comment (dckc) [inline .husky/pre-commit:1]: `npx` involves runnnig code that's not verified against hashes in yarn.lock. Doing this in ci is a risk we do not usually take, I think.

File: package.json [hunk 2]
```diff
@@ -11,7 +11,9 @@
     "test": "ava",
     "lint-fix": "yarn lint --fix",
     "lint": "eslint 'snippets/**/*.js'",
-    "build": "exit 0"
+    "format": "prettier --write '**/*.md' --config .prettierrc.json",
+    "build": "exit 0",
+    "prepare": "husky"
```
Comment (dckc) [inline package.json:16]: I think the shop has avoided pre-commit hooks, to date. Enforcement is done by preventing PRs from merging until they pass ci.

A `prettier --check` in the `lint` script is what I would expect. For precedent, see...

https://github.com/Agoric/agoric-sdk/blob/66e4cc0f36b5ed33f1b3fa235a335757ca868b9c/package.json#L60-L63

File: .prettierrc.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{
+    "printWidth": 80,
+    "tabWidth": 2,
+    "useTabs": false,
+    "semi": false,
```
Comment (dckc) [inline .prettierrc.json:5]: Is this config used to format `.js` as well?

Our [house style](https://github.com/Agoric/agoric-sdk/wiki/Coding-Style) follows Airbnb, which _does_ include semicolons.

See also https://github.com/Agoric/agoric-sdk/tree/master/packages/eslint-config

File: .prettierrc.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    "singleQuote": true,
+    "proseWrap": "preserve",
+    "trailingComma": "none",
```
Comment (dckc) [inline .prettierrc.json:8]: we do trailing commas too

**Review by dckc — COMMENTED** [review]
This is good progress, but please use a lint check rather than precommit hook.

File: .prettierrc.json (hunk 3 above)
Comment (mujahidkay) [reply .prettierrc.json:5]: This is just for markdown files

File: .husky/pre-commit (hunk 1 above)
Comment (mujahidkay) [reply .husky/pre-commit:1]: I'll replace it with a lint check. I wasn't aware of this - thanks for letting me know.

Comment (dckc) [reply .husky/pre-commit:1]: In this case, I suppose `lint-staged` _is_ in `yarn.lock`. But one typo and you're running `lint-stoged` which does who-knows-what.

**Comment by mujahidkay** [comment]
> This is good progress, but please use a lint check rather than precommit hook.

I have addressed pre-commit hook related comments. The others don't apply since we are not messing with js files at all. PTAL.

File: package.json [hunk 5]
```diff
@@ -11,6 +11,8 @@
     "test": "ava",
     "lint-fix": "yarn lint --fix",
     "lint": "eslint 'snippets/**/*.js'",
+    "format": "prettier --write '**/*.md' --config .prettierrc.json",
+    "lint:format": "prettier --check '**/*.md' --config .prettierrc.json",
```
Comment (dckc) [inline package.json:15]: To keep things from decaying, this has to be enforced in ci. So the `lint` script should call `lint:format`, I think. Or the ci workflow should be changed to call `lint:format`.

I suppose we can take care of that in a later PR, but here would be better.

**Review by dckc — APPROVED** [review]
I'd like ci enforcement to go into this PR. But doing it in a later PR is acceptable.

File: package.json (hunk 5 above)
Comment (mujahidkay) [reply package.json:15]: Already added in this PR. https://github.com/Agoric/documentation/blob/a7bb1e108b5b5a9dc2e5a9465906a1ba5837ac08/.github/workflows/test-markdown.yml#L29

**Pull request merged**